Entities/Vehicles/Common/VehicleConvert.as

```diff
@@ -38,145 +73,110 @@ void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
+			CBlob@ b = blobsInSector[i];
 			if (b !is this && b.hasTag("player") && !b.hasTag("dead"))
 			{
 				if (b.getTeamNum() != this.getTeamNum())
 				{
-					Vec2f bpos = b.getPosition();
-					if (bpos.x > pos.x - this.getWidth() / 1.0f && bpos.x < pos.x + this.getWidth() / 1.0f &&
-					        bpos.y < pos.y + this.getHeight() / 1.0f && bpos.y > pos.y - this.getHeight() / 1.0f)
-					{
-						attackersCount++;
-						attackerTeam = b.getTeamNum();
-					}
+					attackersCount++;
+					attackerTeam = b.getTeamNum();
 				}
 				else
 				{
 					friendlyCount++;
 				}
 			}
 		}
+	}
 
-		int ticks = GetCaptureTime(this);
-		if (this.hasTag(raid_tag))
+	if (attackersCount > 0 || this.hasTag(raid_tag))
+	{
+		int ticks = this.get_u16(counter_prop);
+		const u16 captureTime = this.get_u16(capture_time_prop);
+
+		//convert
+		if (attackersCount > friendlyCount)
 		{
-			ticks = this.get_s16(counter_prop);
+			ticks--;
 		}
-
-		if (attackersCount > 0 || ticks < GetCaptureTime(this))
+		//un-convert gradually
+		else if (attackersCount < friendlyCount || attackersCount == 0)
 		{
-			//convert
-			if (attackersCount > friendlyCount)
-			{
-				ticks--;
-			}
-			//un-convert gradually
-			else if (attackersCount < friendlyCount || attackersCount == 0)
-			{
-				ticks = Maths::Min(ticks + 1, GetCaptureTime(this));
-			}
+			ticks = Maths::Min(ticks + 1, captureTime);
+		}
 
-			this.set_s16(counter_prop, ticks);
-			this.Tag(raid_tag);
+		this.set_u16(counter_prop, ticks);
+		this.Tag(raid_tag);
 
-			if (ticks <= 0)
-			{
-				this.server_setTeamNum(attackerTeam);
-				reset_timer = true;
-			}
-			else
+		if (ticks <= 0)
+		{
+			this.server_setTeamNum(attackerTeam);
+			ResetProperties(this);
+		}
+		else
+		{
+			this.set_u16(friendly_prop, friendlyCount);
+			this.set_u16(enemy_prop, attackersCount);
+			
+			if (attackersCount == 0 && ticks >= captureTime)
 			{
-				this.set_s16(friendly_prop, friendlyCount);
-				this.set_s16(enemy_prop, attackersCount);
-				reset_timer = false;
+				this.Untag(raid_tag);
 			}
-
-			sync = true;
 		}
-	}
-	else
-	{
-		this.Untag(raid_tag);
-	}
-
-	if (reset_timer)
-	{
-		this.set_s16(friendly_prop, 0);
-		this.set_s16(enemy_prop, 0);
 
-		this.set_s16(counter_prop, GetCaptureTime(this));
-		this.Untag(raid_tag);
-		sync = true;
+		SyncProperties(this);
 	}
-
-	if (sync)
-	{
-		this.Sync(friendly_prop, true);
-		this.Sync(enemy_prop, true);
-
-		this.Sync(counter_prop, true);
-		this.Sync(raid_tag, true);
-	}
-
 }
 
 void onChangeTeam(CBlob@ this, const int oldTeam)
 {
-	if (this.getTeamNum() >= 0 && this.getTeamNum() < 10)
+	ConvertPoints(this, "VEHICLE,BOW,DOOR");
+
+	if (this.getTeamNum() < 10)
 	{
 		CSprite@ sprite = this.getSprite();
 		if (sprite !is null)
 		{
 			sprite.PlaySound("/VehicleCapture");
 		}
-
-		ConvertPoint(this, "VEHICLE");
-		ConvertPoint(this, "DOOR");
 	}
 }
 
-void ConvertPoint(CBlob@ this, const string pointName)
+void ConvertPoints(CBlob@ this, const string pointNames)
```

Comment:
`pointNames` really should be a `string[]`

Reply:
Resolved, review again? :D